node/node_syncing.go

```diff
@@ -578,6 +578,23 @@ func (node *Node) CalculateResponse(request *downloader_pb.DownloaderRequest, in
 				Msg("[SYNC] extra node registered")
 		}
 
+	case downloader_pb.DownloaderRequest_BLOCKBYHEIGHT:
+		if len(request.Heights) == 0 {
+			return response, errors.New("empty heights list provided")
+		}
+		out := make([][]byte, 0, len(request.Heights))
+		for _, v := range request.Heights {
```

Comment:
Please set a limit on the number of blocks queried to avoid abuse.

Reply:
Consider this PR? https://github.com/harmony-one/harmony/pull/3938/files The PR need to be updated though

Reply:
I've created PR https://github.com/JackyWYX/harmony/pull/2